lib/index.js

```diff
@@ -319,3 +355,114 @@ exports.socket = createSocket;
+
+function linger(number){
+  if(number){
+    if(setopt(this.binding, 'linger', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type + ' linger@' + number + '\n');
+  } else {
+    return getopt(this.binding, 'linger');
+  }
+}
+function sndbuf(number){
+  if(number){
+    if(setopt(this.binding, 'sndbuf', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type + ' sndbuf@' + number + '\n');
+  } else {
+    return getopt(this.binding, 'sndbuf');
+  }
+}
+
+function rcvbuf(number){
+  if(number){
+    if(setopt(this.binding, 'rcvbuf', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type +' rcvbuf@' + number + '\n');
+  } else {
+    return getopt(this.binding, 'rcvbuf');
+  }
+}
+
+function sndtimeo(number){
+  if(number){
+    if(setopt(this.binding, 'sndtimeo', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type + ' reconn@' + number + '\n');
+  } else {
+    return getopt(this.binding, 'sndtimeo');
+  }
+}
+
+function rcvtimeo(number){
+  if(number){
+    if(setopt(this.binding, 'rcvtimeo', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type + ' reconn@' + number + '\n');
+  } else {
+    return getopt(this.binding, 'rcvtimeo');
+  }
+}
+
+function reconn(number){
+  if(number){
+    if(setopt(this.binding, 'reconn', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type + ' reconn@' + number + '\n');
+  } else {
+    return getopt(this.binding, 'reconn');
+  }
+}
+
+function maxreconn(number){
+  if(number){
+    if(setopt(this.binding, 'maxreconn', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type + ' maxreconn@' + number +'\n');
+  } else {
+    return getopt(this.binding, 'maxreconn');
+  }
+}
+
+function sndprio(number){
+  if(number){
+    if(setopt(this.binding, 'sndprio', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type + ' sndprio@' + number + '\n');
+  } else {
+    return getopt(this.binding, 'sndprio');
+  }
+}
+
+function rcvprio(number){
+  if(number){
+    if(setopt(this.binding, 'rcvprio', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type + ' rcvprio@' + number + '\n');
+  } else {
+    return getopt(this.binding, 'rcvprio');
+  }
+}
+
+function tcpnodelay(bool){
+  if(arguments.length){
+    if(bool){
+      if(nn.Setopt(this.binding, nn.NN_TCP, nn.NN_TCP_NODELAY, 1) > -1)
+        return true;
+      throw new Error(nn.Err() + ': '+this.type + ' nodelay@'+'activing'+'\n');
+    } else {
+      if(nn.Setopt(this.binding, nn.NN_TCP, nn.NN_TCP_NODELAY, 0) > -1)
+        return false;
+      throw new Error(nn.Err() + ': '+this.type+' nodelay@'+'deactiving'+'\n');
+    }
+  } else {
+    switch(nn.Getopt(this.binding, nn.NN_TCP, nn.NN_TCP_NODELAY)){
+      case 1: return true;
+      case 0: return false;
+      default:
+        throw new Error(nn.Err() +': '+this.type+' nodelay@'+'getsockopt'+'\n');
```

Comment:
the first two, definitely, but i had to think about the 3rd one: https://github.com/nickdesaulniers/node-nanomsg/blob/sockopts/lib/index.js#L313

since the callback here will fire on some libnanomsg error during an `nn_device()` operation, we can just use `nn.Err()` without needing to carry the param back up into JS